src/plugins/status_page/public/status_page.js

```diff
@@ -36,6 +36,14 @@ const chrome = require('ui/chrome')
         ui.serverState = overall.state;
         ui.serverStateMessage = overall.title;
       }
+
+      // Unless the status page was explicitly requested, there is no
+      // reason to stay on the status page if the status is green; reload
+      // the window so the user is shown the page they were requesting.
+      const statusPageUrl = chrome.addBasePath('/status');
+      if ((overall.state === 'green') && ($window.location.pathname !== statusPageUrl)) {
+        return $window.location.reload();
+      }
```

Comment:
Yeah, I disagree with this approach. The issue here is a race condition caused by legacy code from back when the status page automatically refreshed. I think the right thing to do here is get rid of the race condition and send the status information to the browser with the app code, rather than fetching it on page load.